### PR TITLE
[SwiftSyntax] Simplify AbsolutePosition offset calculation and support columns

### PR DESCRIPTION
**Pull request by harlanhaskins**
This patch simplifies the calculation and structure of AbsolutePosition, in preparation for DiagnosticEngine changes to easily create SourceLocations.

### CODE REVIEW

File: tools/SwiftSyntax/Trivia.swift.gyb [hunk 1]
```diff
@@ -99,59 +99,6 @@ extension TriviaPiece: TextOutputStreamable {
...
-% end
-    }
-  }
-
-  /// Computes the information from this trivia to inform the source locations
-  /// of the associated tokens.
-  /// Specifically, walks through the trivia and keeps track of every newline
-  /// to give a number of how many newlines and UTF8 characters appear in the
-  /// trivia, along with the UTF8 offset of the last column.
-  func characterSizes() -> (lines: Int, lastColumn: Int, utf8Length: Int) {
```
Comment (harlanhaskins) [inline tools/SwiftSyntax/Trivia.swift.gyb]: Surprisingly, this code was never called.

**Comment by harlanhaskins** [comment]
@nkcsgexi I'm wondering if we even need the utf-16 encoding.

**Review by nkcsgexi — APPROVED** [review]
LGTM!

**Comment by harlanhaskins** [comment]
The calculation is a little bit wrong, I'm investigating now.

File: tools/SwiftSyntax/AbsolutePosition.swift [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  public fileprivate(set) var column: Int
+
+  public init(line: Int = 1, column: Int = 1, utf8Offset: Int = 0) {
+    self.line = line
+    self.column = column
+    self.utf8Offset = utf8Offset
+  }
+
+  internal func add(columns: Int) {
+    print("adding \(columns) column bytes")
```
Comment (harlanhaskins) [inline tools/SwiftSyntax/AbsolutePosition.swift]: Will remove before merging

**Comment by harlanhaskins** [comment]
Wow, very simple bug.

@swift-ci please smoke test

**Pull request merged**